tests/test_sqlackqueue.py

```diff
@@ -0,0 +1,314 @@
+
+        def consumer():
+            for _ in range(100):
+                data = queue.get(block=True)
+                self.assertTrue('var' in data)
+
+        c = Thread(target=consumer)
+        c.start()
+        producers = []
+        for seq in range(10):
+            t = Thread(target=producer, args=(seq,))
+            t.start()
+            producers.append(t)
+
+        for t in producers:
+            t.join()
+
+        c.join()
+
+    def test_multiple_consumers(self):
+        """Test sqlqueue can be used by multiple consumers."""
+
+        queue = SQLiteAckQueue(path=self.path, multithreading=True,
+                            auto_commit=self.auto_commit)
+
+        def producer():
+            for x in range(1000):
+                queue.put('var%d' % x)
+
+        counter = []
+        # Set all to 0
+        for _ in range(1000):
+            counter.append(0)
+
+        def consumer(index):
+            for i in range(200):
+                data = queue.get(block=True)
+                self.assertTrue('var' in data)
+                counter[index * 200 + i] = data
+
+        p = Thread(target=producer)
+        p.start()
+        consumers = []
+        for index in range(5):
+            t = Thread(target=consumer, args=(index,))
+            t.start()
+            consumers.append(t)
+
+        p.join()
+        for t in consumers:
+            t.join()
+
+        self.assertEqual(0, queue.qsize())
+        for x in range(1000):
+            self.assertNotEqual(0, counter[x],
+                                "not 0 for counter's index %s" % x)
+
+    def test_task_done_with_restart(self):
+        """Test that items are not deleted before task_done."""
+
+        q = SQLiteAckQueue(path=self.path,
+                           # auto_commit=False
+                           )
+
+        for i in range(1, 11):
+            q.put(i)
+
+        self.assertEqual(1, q.get())
+        self.assertEqual(2, q.get())
+        # size is correct before task_done
+        self.assertEqual(8, q.qsize())
+        q.task_done()
+        # make sure the size still correct
+        self.assertEqual(8, q.qsize())
+
+        self.assertEqual(3, q.get())
+        # without task done
+        del q
+#         q = SQLiteAckQueue(path=self.path,
+#                            # auto_commit=False
+#                            )
+#         # After restart, the qsize and head item are the same
+#         self.assertEqual(8, q.qsize())
+#         # After restart, the queue still works
+#         self.assertEqual(3, q.get())
+#         self.assertEqual(7, q.qsize())
+
+    def test_protocol_1(self):
+        shutil.rmtree(self.path, ignore_errors=True)
+        q = SQLiteAckQueue(path=self.path)
+        self.assertEqual(q.protocol, 2 if sys.version_info[0] == 2 else 4)
+
+    def test_protocol_2(self):
+        q = SQLiteAckQueue(path=self.path)
+        self.assertEqual(q.protocol, None)
+
+
+class SQLite3QueueNoAutoCommitTest(SQLite3QueueTest):
```

Comment:
Looks like AckQueue doesn't support `autocommit=False`, if YES, the test cases should be removed as well